src/agent/src/main.rs

```diff
@@ -427,6 +431,17 @@ fn init_attestation_components(logger: &Logger, _config: &AgentConfig) -> Result
     )
     .map_err(|e| anyhow!("launch_process {} failed: {:?}", AA_PATH, e))?;
 
+    let ocicrypt_config = serde_json::json!({
+        "key-providers": {
+            "attestation-agent":{
+                "ttrpc":CDH_URI
+            }
+        }
+    });
+
+    fs::write(OCICRYPT_CONFIG_PATH, ocicrypt_config.to_string().as_bytes())?;
+    env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", OCICRYPT_CONFIG_PATH);
```

Comment:
would it be an option to move this (writing the file and setting the env) down the stack to image-rs?

Reply:
This might be a choice. But as a crate, `image-rs` wants the config env be passed via higher binary rather than set by itself.

I get your point setting such an env in kata-agent would make some potential trouble. A better solution to me is we can have CDH set this after moving image pull stuff to CDH fully per https://github.com/kata-containers/kata-containers/issues/9266

For now, as this way is used in CCv0, probably it makes sense to keep the workaround still until better solution is brought?

Reply:
> This might be a choice. But as a crate, image-rs wants the config env be passed via higher binary rather than set by itself.

Can you elaborate a bit? not sure I understand. Since ocicrypt-rs is an implementation detail of image-rs, wouldn't it be possible to set the env locally in image-rs like the [units tests](https://github.com/confidential-containers/guest-components/blob/main/image-rs/src/decrypt.rs#L301-L304)?

This way we don't have to leak image-rs specifics to the kata codebase.

Reply:
`ocicrypt-rs` is a rust version of [ocicrypt](https://github.com/containers/ocicrypt), and `image-rs` is a rust version of [image](https://github.com/containers/image). The two are both libs and I think the setting thing should not also be added to image-rs.

Another reason is for [enclave-cc](https://github.com/confidential-containers/enclave-cc/blob/main/src/enclave-agent/src/config.rs#L39), the config path is set inside `image-rs`'s caller. If we set env inside image-rs, the original one would be overwritten.

I get your point that we'd better keep kata codebase away from image stuff. So my idea is to leave it as-is in this PR. Later, when we decided to move the image-pulling things fully to confidential data hub, kata-agent could have a very clean codebase without any image pull stuff even `image-rs`. Those things would all be implemented inside CDH. 

wdyt?

Reply:
If it's a requirement of enclave-cc, then this would be a reason to do this here. maybe there's an alternative that doesn't taint kata code directly, like the env could be set outside of kata the process `OCICRYPT_...=/etc/ocicrypt... kata-agent` and the config file added statically to the rootfs, not sure how/whether this would work in the current rootfs build process, though.

In general, I still think that ocicrypt-rs is a dep of image-rs (i.e. an impl detain that could change in the future) and hence kata has no business configuring it directly, but maybe i'm overly neurotic and others can chime in. cc @fidencio

Reply:
Yea I think set outside kata process is a right way to go for me. As the CDH socket is fixed, we could do this. If in non-coco scenario, the env actually will not be used. Technically it works for systemd, I am not sure whether init process works. Also I want to hear more voices upon this.

Reply:
>  mkulke: ...like the env could be set outside of kata the process OCICRYPT_...=/etc/ocicrypt... kata-agent and the config file added statically to the rootfs, not sure how/whether this would work in the current rootfs build process, though.

I agree with both these points: (1) make the config file statically part of the rootfs, assuming the build process isn't difficult, and (2) set the environment variable outside of kata-agent. And yet...

> Xynnn007: ... A better solution to me is we can have CDH set this after moving image pull stuff to CDH fully per https://github.com/kata-containers/kata-containers/issues/9266

--so if this change to `init_attestation_components` is temporary, it seems OK.

Reply:
Set env apart from both kata-agent and CDH might be the finest way for decoupling. some thoughts:
1. For guest images where `AGENT_INIT=false`, we can set the env in systemd-service config
2. For initrds where `AGENT_INIT=true`. How can we set the env for `/sbin/init`?

cc @fidencio